packages/http/src/mocker/serializer/style/deepObject.ts

```diff
@@ -0,0 +1,19 @@
+import { HttpParamStyles } from '@stoplight/types';
+
+export function serializeWithDeepObjectStyle(name: string, value: any) {
+  return serialize(name, [], value);
+}
+
+function serialize(name: string, path: string[], value: any): string {
+  if (typeof value === 'object') {
+    return Object.keys(value)
+      .map(key => serialize(name, [...path, isPositiveInteger(key) ? '' : key], value[key]))
+      .join('&');
+  } else {
+    return `${name}${path.map(key => `[${key}]`).join('')}=${value}`;
+  }
+}
+
+function isPositiveInteger(str: string) {
+  return /^\+?\d+$/.test(str) && parseInt(str) >= 0;
```

Comment:
Do we need the regex check here?

Reply:
Actually it looks like the shortest way of checking for a positive integer. Other ways include casting, checking for (-)Infinity, NaNs, dirty stuff basically.

Reply:
Note: we are looking for a positive integer inside a string.